### PR TITLE
As a CAS user, I want to see user information when approving prime admins #869

### PR DESCRIPTION
**Pull request by marcellmueller**
Implements #869

### CODE REVIEW

File: client/app/components/form/UserOperatorReviewForm.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+          />
+        ),
+      }}
+      // If the operator is new, the first section should be expanded
+      // If the user is pending, the second section should be expanded
+      expandedSteps={{
+        "Operator Information": isNewOperator,
+        "User Information": userOperatorStatus === "Pending",
+      }}
+    />
```
Comment (marcellmueller) [inline client/app/components/form/UserOperatorReviewForm.tsx:54]: Note: this was not in the acceptance criteria so happy to remove this if you think it's a bad idea. This expands each section by default if we are displaying the review component.

File: client/app/components/datagrid/OperationDataGrid.tsx [hunk 2]
```diff
@@ -42,7 +47,7 @@ const OperationDataGrid = ({
             {/* 🔗 Add reg or details link */}
             <Link
               className="no-underline text-bc-link-blue whitespace-normal"
-              href={`operations/${params.row.id}/1`}
+              href={`operations/${params.row.id}${isIndustryUser ? "/1" : ""}`}
```
Comment (marcellmueller) [inline client/app/components/datagrid/OperationDataGrid.tsx:50]: This is unrelated to this PR though I noticed it was broken. I'm unsure of what happened here as this was previously implemented. Likely a rebase issue.

**Comment by shon-button** [comment]
Having issues with the API.  
```
[14/Feb/2024 15:34:27] "GET /api/registration/user-operators HTTP/1.1" 401 26
[14/Feb/2024 15:34:40] "GET /api/registration/operations HTTP/1.1" 401 26
```
Any insight to help resolve?

**Comment by marcellmueller** [comment]
> After `make reset_db && make run`, having issues with the API as `cas_admin`
> 
> ```
> [14/Feb/2024 15:34:27] "GET /api/registration/user-operators HTTP/1.1" 401 26
> [14/Feb/2024 15:34:40] "GET /api/registration/operations HTTP/1.1" 401 26
> ```
> 
> Any insight to help resolve? Thanks.

Hey I've been having similar issues recently though I don't think it's related to this PR. I just replicated it on develop as well. I would almost suspect it's due to the new user caching though that explanation doesn't make sense since we are doing `make reset_db`.

If I do `make reset_db && make run` I get errors even after logging out and signing back in. Though if I reset my backend again and try again I get authorization.

I can fix it by running `make reset_db && make run`, logging out and back in again to create user, then what seems to fix it is shutting down and restarting the backend just using `make run` this time.

File: client/app/components/form/UserOperatorReviewForm.tsx (hunk 1 above)
Comment (shon-button) [reply client/app/components/form/UserOperatorReviewForm.tsx:54]: IMO, it's a great idea!

**Comment by shon-button** [comment]
👌looks really great!

For your consideration, for use case where both `operator` and `user operator` require approval, can we reset the `review component` if the `user operator` "Approval" fails because of approval sequence dependence on `operator`  "Approval"? 
Example, if I approve the "User Information" and get an error message about needing to approve the Operator first I cannot do so unless I refresh the page as the  "User Information" approval button is gone.

**Review by shon-button — APPROVED** [review]

**Comment by shon-button** [comment]
> > 👌looks really great!
> > For your consideration, for use case where both `operator` and `user operator` require approval, can we reset the `review component` if the `user operator` "Approval" fails because of approval sequence dependence on `operator` "Approval"? Example, if I approve the "User Information" and get an error message about needing to approve the Operator first I cannot do so unless I refresh the page as the "User Information" approval button is gone.
> 
> Thanks this was a good suggestion. I've made a commit to solve this issue, on success I am generating a new random key which is passed into the `key` prop of the prime admin review form. Generating a new key forces a rerender and resets the review component. Can you give it a test for me before I merge this in?
> 
> [Commit here](https://github.com/bcgov/cas-registration/pull/903/commits/3f3454659615a7c484ddfb134d5252c0b97bbb61)

👀 I was able to approve the user before the operator:
**Steps**
- `bc-cas-dev-three` create new operator
- `cas-admin` view new operator in screen `operators`
- `cas-admin`  open new operator and approve user
image
